nimble/host/mesh/src/access.c

```diff
@@ -494,11 +501,6 @@ void bt_mesh_model_recv(struct bt_mesh_net_rx *rx, struct os_mbuf *buf)
 			if (elem->addr != rx->dst) {
 				continue;
 			}
-		} else if (BT_MESH_ADDR_IS_GROUP(rx->dst) ||
-			   BT_MESH_ADDR_IS_VIRTUAL(rx->dst)) {
-			if (!bt_mesh_elem_find_group(elem, rx->dst)) {
-				continue;
-			}
 		} else if (i != 0 || !bt_mesh_fixed_group_match(rx->dst)) {
```

Comment:
This is now broken. `bt_mesh_fixed_group_match()` is supposed to be used when you already know the address is a fixed group one. Now that you removed the intermediate checks for `IS_GROUP()` and `IS_VIRTUAL()` those types of addresses will get passed to `bt_mesh_fixed_group_match()` which will always return false for them, causing messages to be ignored before `find_op()` has a chance to be called. What I meant when I said you should remove the redundant check is not just removing the middle `else if` branch, but moving the adress type checks to the last branch so that `fixed_group_match()` is still only called for fixed group addresses.